src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -2,23 +2,207 @@
+
+            for (var i = 0; i < singleParameters.Length; i++)
+            {
+                Parameter parameter = singleParameters[i];
+                if (parameter.Length == null ||
+                    parameter.Type is not CSPointer ptr || ptr.BaseType is not CSType baseType)
+                {
+                    continue;
+                }
+
+                string typeName = "Vector" + vectorSize + vectorType;
+                string firstElem = ".X";
+                if (!_existingVectorTypes.Contains(typeName))
+                {
+                    typeName = baseType.TypeName;
+                    firstElem = "";
+                }
+
+                Parameter? refParam = null;
+                Constant constant;
+                if (parameter.Length is Constant cnst)
+                {
+                    constant = cnst;
+                }
+                else if (parameter.Length is BinaryOperation binary)
+                {
+                    ParameterReference? reference = null;
+                    if (binary.Left is ParameterReference leftRef && binary.Right is Constant rightConst)
+                    {
+                        reference = leftRef;
+                        constant = rightConst;
+                    }
+                    else if (binary.Right is ParameterReference rightRef && binary.Left is Constant leftConst)
+                    {
+                        reference = rightRef;
+                        constant = leftConst;
+                    }
+                    else continue;
+
+                    refParam = singleParameters.First(p => p.Name == reference.ParameterName);
+                    countParameters.Add(refParam);
+                }
+                else continue;
+
+                if (constant.Value == vectorSize)
+                {
+                    CSType vector = new CSType(typeName, baseType.Constant);
+                    singleParameters[i] = parameter with
+                    {
+                        Type = new CSRef(ptr.Constant || baseType.Constant ? CSRef.Type.In : CSRef.Type.Ref,
+                            vector),
+                        Length = null
+                    };
+                    spanParameters[i] = parameter with { Type = new CSSpan(vector, ptr.Constant || baseType.Constant), Length = null };
+                    arrayParameters[i] = parameter with { Type = new CSArray(vector, ptr.Constant || baseType.Constant), Length = null };
+                    nameTable.Rename(parameter, parameter.Name + "_ptr");
+                    overloadedParams.Add((parameter, singleParameters[i], firstElem));
+                }
+            }
+
+            if (overloadedParams.Count == 0)
+            {
+                newOverloads = null;
+                return false;
+            }
+
+            string overloadName = overload.OverloadName.Remove(m.Index + 2, 1);
```

Comment:
Here we are relying on the vector stuff being at the end of the function name, so that suggests to me that we want to add the `$` to the name matching regex above.

Reply:
This is not relying on the vector stuff being at the end? This works perfectly fine if its in the middle too. However it doesn't work if there are multiple matches in the name.

Reply:
Yeah sorry, I read this wrong.